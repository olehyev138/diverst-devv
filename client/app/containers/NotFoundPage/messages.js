/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'diverst.containers.NotFoundPage';

export default defineMessages({
  header: {
    id: `${scope}.headers.header`,
  },
});
