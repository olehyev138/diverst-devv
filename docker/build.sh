#!/bin/bash

# Set path from second argument or default to current path
BUILD_PATH=${2:-.}

if [ -n "$1" ]; then
  docker build -t "$1" "$BUILD_PATH" "${@:3}"
else
  echo "ERROR: Requires a docker image tag as first argument (such as diverstdev:X.X)" 1>&2
  exit 1
fi
