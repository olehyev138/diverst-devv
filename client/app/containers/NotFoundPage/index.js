/*
 * LoginPage
 *
 * This is the first thing users see of our App, at the "/login" route
 */

import React from "react";
import { FormattedMessage } from "react-intl";

import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";

import messages from "./messages";

import logo from "images/diverst.png";

/* eslint-disable react/prefer-stateless-function */
export class NotFoundPage extends React.PureComponent {

    render() {
        return (
        <div>
          <Grid container spacing={0} direction="column" alignItems="center" justify="center" style={{minHeight: '100vh', textAlign: "center"}}>
            <Grid item xs={6}>
              <img src={logo} alt="Diverst Logo" height="150" width="150"/>
              
              <h1><FormattedMessage {...messages.header} /></h1>
              <Button>Home</Button>
            </Grid>   
          </Grid>
        </div>
        );
    }
}

export default NotFoundPage;