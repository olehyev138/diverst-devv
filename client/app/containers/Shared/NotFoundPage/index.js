/**
 * NotFoundPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import WrappedNavLink from 'components/Shared/WrappedNavLink';

import {
  Button, Typography, Divider, Hidden
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import HomeIcon from '@material-ui/icons/Home';

import { ROUTES } from 'containers/Shared/Routes/constants';

import messages from 'containers/Shared/NotFoundPage/messages';
import Logo from 'components/Shared/Logo';

const styles = theme => ({
  textCenter: {
    textAlign: 'center',
  },
  verticalMargins: {
    marginTop: 24,
    marginBottom: 24,
  },
});

export class NotFoundPage extends React.PureComponent {
  render() {
    const { classes } = this.props;

    return (
      <div className={classes.textCenter}>
        <Logo coloredDefault imgClass='extra-large-img' />
        <Typography variant='h6' className={classes.verticalMargins}>
          <FormattedMessage {...messages.header} />
        </Typography>
        <Divider />
        <Button
          variant='contained'
          color='primary'
          component={WrappedNavLink}
          to={ROUTES.user.home.path}
          className={classes.verticalMargins}
        >
          <Hidden xsDown>
            <HomeIcon />
          </Hidden>
          Return To Home
        </Button>
      </div>
    );
  }
}

NotFoundPage.propTypes = {
  classes: PropTypes.object,
};

export default withStyles(styles)(NotFoundPage);
