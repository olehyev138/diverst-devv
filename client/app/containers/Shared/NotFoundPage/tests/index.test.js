import React from 'react';
import { shallow } from 'enzyme/build';
import { unwrap } from '@material-ui/core/test-utils/index';

import NotFoundPage from 'containers/Shared/NotFoundPage/index';
const NotFoundPageNaked = unwrap(NotFoundPage);

describe('<NotFoundPage />', () => {
  it('should not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<NotFoundPageNaked classes={{}} />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const wrapper = shallow(<NotFoundPageNaked classes={{}} />);

    expect(wrapper).toMatchSnapshot();
  });
});
