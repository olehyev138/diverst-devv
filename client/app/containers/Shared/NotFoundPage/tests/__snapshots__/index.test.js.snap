// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render and match the snapshot 1`] = `
<div>
  <Connect(WithStyles(Logo))
    coloredDefault={true}
    imgClass="extra-large-img"
  />
  <WithStyles(ForwardRef(Typography))
    variant="h6"
  >
    <FormattedMessage
      id="diverst.containers.NotFoundPage.headers.header"
      values={Object {}}
    />
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(Button))
    color="primary"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    to="/"
    variant="contained"
  >
    <Hidden
      xsDown={true}
    >
      <HomeIcon />
    </Hidden>
    Return To Home
  </WithStyles(ForwardRef(Button))>
</div>
`;
